src/status_im/mailserver/core.cljs

```diff
@@ -276,9 +276,19 @@
     (log/debug "Adjusting mailserver request" "from:" from "adjusted-from:" adjusted-from)
     adjusted-from))
 
-(fx/defn handle-request-success [{:keys [db]} {:keys [request-id]}]
+(fx/defn handle-request-success [{{:keys [chats] :as db} :db}
+                                 {:keys [request-id topics]}]
   (when (:mailserver/current-request db)
-    {:db (assoc-in db [:mailserver/current-request :request-id] request-id)}))
+    (if (= 1 (count topics))
```

Comment:
This is a bit fragile and it's not immediately clear what we are trying to do, probably I would factor it in a function:
`(initial-public-chat-request? ...)` so it's clearer what we are trying to do (also tests :) )

Reply:
I see why this is fragile - this is it, the piece I was missing 😁. Thanks for the hint! - had the whole mental model incomplete(still ways to go, I'm sure :p). Going to update accordingly.

Reply:
I made the naming in let binding clear instead of factoring it in a function. Hope that is acceptable.

Otherwise, this was indeed the missing piece. I had only observed that the topics in request was one in most cases before. Didn't realize that the request could have multiple topics like in the case of handling sync of public chats between devices.